app/payments.py

```diff
@@ -126,61 +126,55 @@ def get_payments(id):
         result['error'] = result['error'].format(id)
         rc = HTTP_404_NOT_FOUND
 
+
     return make_response(jsonify(result), rc)
 
 ######################################################################
 # UPDATE AN EXISTING PAYMENT
 ######################################################################
 @app.route('/payments/<int:id>', methods=['PUT'])
 def update_payments(id):
-    index = [i for i, payment in enumerate(payments) if payment['id'] == id]
-    if len(index) > 0:
+    try:
+        if not request.data:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
         if not request.is_json:
-            return make_response(CONTENT_ERR_MSG, HTTP_400_BAD_REQUEST)
-        payload = request.get_json()
-        if is_valid(payload):
-            payments[index[0]] = {'id' : id, 'nickname' : payload['nickname'], 'type' : payload['type'],
-                                  'default' : payments[index[0]]['default'],
-                                  'charge-history' : payments[index[0]]['charge-history'],
-                                  'detail' : payload['detail']}
-            message = payments[index[0]]
-            rc = HTTP_200_OK
-        else:
-            message = { 'error' : 'Payments data was not valid' }
-            rc = HTTP_400_BAD_REQUEST
-    else:
-        message = { 'error' : 'Payments %s was not found' % id }
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
+        try:
+            data = request.get_json()
+        except Exception:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
+        message = payment_service.update_payment(id,payment_replacement=data)
+        rc = HTTP_200_OK
+    except InvalidPaymentID as e:
+        message = 'Invalid payment: Payment ID not found'
         rc = HTTP_404_NOT_FOUND
-
+    except DataValidationError as e:
+        message = 'Invalid payment: body of request contained bad or no data'
+        rc = HTTP_400_BAD_REQUEST
```

Comment:
That does work, but it fails when mocking even if i mock the interface layer's exception with a proper message, will look into it in further detail and make changes